media/js/firefox/features-sync.js

```diff
@@ -1,104 +1,126 @@
+
+                    // Query if the UITour API is working before we use the API
+                    Mozilla.UITour.getConfiguration('sync', function (config) {
+
+                        // Variation #1: Firefox 31+ signed IN to Sync (default)
+                        if (config.setup) {
+                            SyncPage.setBodyClass('state-fx-31-signed-in');
+                            SyncPage.state = 'Firefox 31 or Higher: Signed-In';
+
+                        // Variation #2: Firefox 31+ signed OUT of Sync
+                        } else {
+                            SyncPage.setBodyClass('state-fx-31-signed-out');
+                            SyncPage.state = 'Firefox 31 or Higher: Signed-Out';
+                        }
+
+                        // Call GA tracking here to ensure it waits for the
+                        // getConfiguration async call
+                        SyncPage.trackPageState(SyncPage.state);
                     });
-                });
 
-            // Variation #3: Firefox 30 or older
-            } else if (fxMasterVersion <= 30) {
-                swapState('state-fx-30-older');
-                state = 'Firefox 30 or older';
+                // Variation #3: Firefox 30 or older
+                } else if (SyncPage.fxVersion <= 30) {
+                    SyncPage.setBodyClass('state-fx-30-older');
+                    SyncPage.state = 'Firefox 30 or older';
+                }
             }
+        // Variation #5: Not Firefox
+        } else {
+            SyncPage.setBodyClass('state-not-fx');
+            SyncPage.state = 'Not Firefox';
         }
-    // Variation #5: Not Firefox
-    } else {
-        swapState('state-not-fx');
-        state = 'Not Firefox';
-    }
-
-    // Send page state to GA if it hasn't already been sent in the
-    // getConfiguration callback
-    if (syncCapable === false) {
-        window.dataLayer.push({
-            'event': 'page-load',
-            'browser': state
-        });
-    }
 
-    // Firefox Sync sign in flow button
-    $('#cta-sync').on('click', function(e) {
-        e.preventDefault();
+        // Send page state to GA if it hasn't already been sent in the
+        // getConfiguration callback
+        // Called for all variations *except* Fx 31+
+        if (SyncPage.syncCapable === false) {
+            SyncPage.trackPageState(SyncPage.state);
+        }
 
-        window.dataLayer.push({
-            'event': 'sync-click',
-            'browser': state
-        });
+        SyncPage.ctaSyncClick = function(e) {
+            e.preventDefault();
+
+            SyncPage.dataLayer.push({
+                'event': 'sync-click',
+                'browser': SyncPage.state
+            });
+
+            Mozilla.UITour.showFirefoxAccounts(SyncPage.params.utmParamsFxA());
+        };
+
+        // Firefox Sync sign in flow button
+        // Only displayed for Fx 31+
+        SyncPage.ctaSync.addEventListener('click', SyncPage.ctaSyncClick);
```

Comment:
If this button is shown only for Fx 31+ should we only bind the event if that conditional is true?